### DISCUSSION TITLE
Button Groups and Disclosure Accessibility

### DISCUSSION THREAD

**Discussion by seanmateer** [twilio-labs/paste#1202] [Office hours] [answered]
**Button Groups**
We have a use case for a [buttons group/segmented control](https://twilio.invisionapp.com/share/5XZREP52NS3#/440813623_1-1_-_Overview) component in Email Insights, similar to the button groups used in Msg Insights (which I believe is still using legacy console components for this). I saw the [previous Q](https://github.com/twilio-labs/paste/discussions/903) on this topic, and noted that it since pushed back in the timeline, so my question is around whether the current button components could be hammered into a button group (e.g. remove padding and augment border radius), or if it would be better to rely on a legacy SG component that we apply some paste styling tokens too? 

**Disclosure Accessibility**
We are exploring a potential use case for a dynamic view (screenshot below) within a disclosure powered by the above button groups. I had questions about the whether there are pattern or accessibility limits of what you can stick in a disclosure (any concerns with dynamic view, chart, etc?) 

<img width="1747" alt="Screen Shot 2021-02-17 at 2 43 16 PM" src="https://user-images.githubusercontent.com/4806849/108271778-9197a180-712e-11eb-92f5-8159ce6f2063.png">

**Comment by richbachman** [comment]
Hi @seanmateer 

For the button group, there's no way to force the current Paste buttons into working. Composing a custom button group out of Box is the best bet here. I started a [CodeSandbox](https://codesandbox.io/s/buttongroup-composition-3fqyf?file=/src/index.tsx) to show how this might be done. There's still a lot to flush out, i.e. hover states, pressed states, etc, but it should give you an idea of what might be involved. 

As for the disclosure, we designed/developed that in a way that you can put any content inside. The only real concern is not nesting focusable elements in the disclosure heading: https://paste.twilio.design/components/disclosure#accessibility

I think the biggest accessibility concern is making sure something is announced to users when the content is changed via the button press.

**Comment by SiTaggart** [comment, marked as answer]
For the Button groups what you'll need is actually the Radio button group as you can only select one at a time and you need the ability to communicate selected state correctly.

That's not on our road map for a while though by the looks of things, maybe May? https://paste.twilio.design/roadmap/

You are probably going to hate this, but I'm going to say it anyway: 

Have you considered Tabs? Main justification is that you're just switching two views of content.
